draft-rescorla-tls-subcerts.md

```diff
@@ -308,45 +282,49 @@ While X.509 forbids end-entity certificates from being used as issuers for other
 certificates, it is perfectly fine to use them to issue other signed objects as
 long as the certificate contains the digitalSignature key usage (RFC5280 section
 4.2.1.3). We could define a new signed object format that would encode only the
-semantics that are needed for this application.  For example, the TLS
-`digitally-signed` structure could be used:
+semantics that are needed for this application. This format is intended to be
+very restrictive and non-extensible:
 
 ~~~~~~~~~~
-digitally-signed struct {
-  uint64 notBefore;
+struct {
   uint64 notAfter;
   SignatureScheme algorithm;
-  ServerName serverName;
-  opaque publicKey<0..2^24-1>;
+  opaque publicKey<0..2^16-1>;
 } DelegatedCredential;
 ~~~~~~~~~~
 
-The ServerNameList can be defined as follows (similar to RFC 6066 section 3):
-~~~~~~~~~~
-struct {
-  NameType name_type;
-  select (name_type) {
-    case host_name: HostName;
-  } name;
-} ServerName;
+The encoding of `SignatureScheme`, publicKey and notAfter are the same as in TLS.
 
-enum {
-  host_name(0), (255)
-} NameType;
+The signature of the DelegatedCredential is computed as the concatenation of:
 
-opaque HostName<1..2^16-1>;
-~~~~~~~~~~
+- A context string
+- A single 0 byte which serves as the separator
+- Hash of the X.509 certificate used to sign the DelegatedCredential
+- uint64 notAfter in Big endian format
+- SignatureScheme
+- public key
+
+The context string depends on the TLS version. For TLS versions the context string
```

Comment:
Another possibility is to disallow this for SSL versions and only allow this for TLS 1.2 and higher.